src/test/ui/rfc-2008-non-exhaustive/uninhabited/match.rs

```diff
@@ -0,0 +1,19 @@
+// aux-build:uninhabited.rs
+#![feature(never_type)]
+
+extern crate uninhabited;
+
+use uninhabited::{
+    UninhabitedEnum,
+};
+
+struct A;
+
+// This test checks that a empty match on a non-exhaustive uninhabited type from an extern crate
+// will not compile.
+
+fn cannot_empty_match_on_empty_enum_to_anything(x: UninhabitedEnum) -> A {
+    match x {} //~ ERROR non-exhaustive patterns
```

Comment:
The general structure here is good, but you should be able to extend this with:

```rust
fn $make_name_up(x: UninhabitedStruct/UninhabitedTupleStruct/...) -> A {
    match x {}
}
```
and same for the subsequent files that use `match x {}`.

Also consider:

```rust
#![feature(exhaustive_patterns)]

enum V {}
struct A(V);

fn foo(x: A) -> String {
    match x {}
}
```

In same-crate, due to the compiler not doing a structural analysis of uninhabitedness without `exhaustive_patterns`, you will get errors as well. That's fine, just split those parts out to a separate compile-fail test. The most important thing getting errors in cross-crate.

(The motivation for this is still redundancy and future proofing, i.e. if and when we stabilize `exhaustive_patterns`, it is good to know that we've already covered our basis with these tests wrt. `#[non_exhaustive]` so that accidents don't happen)

Reply:
Added a whole bunch of tests.